hal/common/retarget.cpp

```diff
@@ -477,24 +479,126 @@ extern "C" WEAK void __cxa_pure_virtual(void) {
 
 #endif
 
+typedef struct {
+    uint32_t size;
+} alloc_info_t;
+
+static SingletonPtr<rtos::Mutex> malloc_stats_mutex;
+static mbed_stats_heap_t heap_stats = {0, 0, 0, 0, 0};
+
+void mbed_stats_heap_get(mbed_stats_heap_t *stats)
+{
+    malloc_stats_mutex->lock();
+    memcpy(stats, &heap_stats, sizeof(mbed_stats_heap_t));
+    malloc_stats_mutex->unlock();
+}
+
 #if defined(TOOLCHAIN_GCC)
 #ifdef   FEATURE_UVISOR
 #include "uvisor-lib/uvisor-lib.h"
 #endif/* FEATURE_UVISOR */
 
 #ifndef  FEATURE_UVISOR
 extern "C" {
+
+extern "C" void __malloc_lock( struct _reent *_r );
+extern "C" void __malloc_unlock( struct _reent *_r );
+
 void * __wrap__malloc_r(struct _reent * r, size_t size) {
     extern void * __real__malloc_r(struct _reent * r, size_t size);
+#if !defined(MBED_CONF_CORE_STATS_HEAP) || !MBED_CONF_CORE_STATS_HEAP
```

Comment:
Hi @bogdanm, this is intentional. The uvisor feature will redirect malloc and related functions so they are secure. It will also be providing its own metrics.